sky/skylet/providers/azure/config.py

```diff
@@ -126,9 +129,21 @@ def _configure_resource_group(config):
         .properties.outputs
     )
 
+    # We should wait for the NSG to be created before opening any ports
+    # to avoid overriding the newly-added NSG rules.
+    nsg_id = outputs["nsg"]["value"]
+    nsg_name = nsg_id.split("/")[-1]
+    network_client = NetworkManagementClient(credentials, subscription_id)
```

Comment:
Curious: why not use the same client? Worth a comment.

Reply:
This is a `NetworkManagementClient`, while the previous client is `ResourceManagementClient`. It is not "the same client"?